src/lib/utils/mergeSearchParameters.ts

```diff
@@ -103,6 +105,23 @@ const mergeHierarchicalFacetsRefinements: Merger = (left, right) =>
     },
   });
 
+const mergeRuleContexts: Merger = (left, right) => {
+  const ruleContexts: string[] = uniq(
+    ([] as any)
+      .concat(left.ruleContexts)
+      .concat(right.ruleContexts)
+      .filter(Boolean)
```

Comment:
why do you need to filter empty values out here?

Reply:
`right.ruleContexts` might be `undefined`.